app/server/logging/jsondb/src/main/java/io/syndesis/server/logging/jsondb/controller/ActivityTrackingController.java

```diff
@@ -288,7 +318,8 @@ private void processEventQueue() {
                     }
 
                     // Write the batch..
-                    jsonDB.update("/activity", Json.writer().writeValueAsBytes(batch));
+//                    jsonDB.update("/activity", Json.writer().writeValueAsBytes(batch));
+                    writeBatch(batch);
```

Comment:
so if I understand correctly, you are inserting directly into the DB here, bypassing jsondb. Wouldn't it make more sense to add a 'writeActivities()' method to jsondb and call it ? This would free this class from having internal knowledge of jsondb table structure and handling.

Reply:
Not really.  The writeBatch method is doing stuff that is not generic.  It would be ugly to push higher 'app' level semantics into lower level generic apis.  I know its also not nice to break through the Json DB api abstractions here, but I'm only doing it to improve performance.

The reason that jsondb does not expose something similar right now is that this use pattern if used improperly can lead to inconsistent db state.  Example being inserting records that look like:
```
/a/b/ = "Hello"
/a/b/c/ ="World"
```

The exposed update method double checks those cases, and would delete the `/a/b/` record when the `/a/b/c/` record is added.  That double checking is partly what was slowing us down.